gulpfile.js

```diff
@@ -148,11 +151,46 @@ function prebidSource(webpackCfg) {
   const analyticsSources = helpers.getAnalyticsSources();
   const moduleSources = helpers.getModulePaths(externalModules);
 
-  return gulp.src([].concat(moduleSources, analyticsSources, 'src/prebid.js'))
+  return gulp.src([].concat(moduleSources, analyticsSources, helpers.getPrecompiledPath('src/prebid.js')))
     .pipe(helpers.nameModules(externalModules))
     .pipe(webpackStream(webpackCfg, webpack));
 }
 
+const PRECOMP_TASKS = new Map();
+function babelPrecomp({distUrlBase = null, disableFeatures = null, dev = false} = {}) {
+  if (dev && distUrlBase == null) {
+    distUrlBase = argv.distUrlBase || '/build/dev/'
+  }
+  const key = `${distUrlBase}::${disableFeatures}`;
+  if (!PRECOMP_TASKS.has(key)) {
+    const babelConfig = require('./babelConfig.js')({
+      disableFeatures: disableFeatures ?? helpers.getDisabledFeatures(),
+      prebidDistUrlBase: distUrlBase ?? argv.distUrlBase
+    });
+    const precompile = function() {
+      return gulp.src(SOURCE_PAT, {base: '.', since: gulp.lastRun(precompile)})
+        .pipe(sourcemaps.init())
+        .pipe(babel(babelConfig))
+        .pipe(sourcemaps.write('.'))
+        .pipe(gulp.dest(helpers.getPrecompiledPath()));
+    }
+    PRECOMP_TASKS.set(key, precompile)
+  }
+  return PRECOMP_TASKS.get(key);
+}
+
+function copyVerbatim() {
```

Comment:
Why are those files copied? To pipe them were?

Reply:
The build generates "precompiled" `.js` and `.d.ts` files under `dist/src`, which is also what's exported from `package.json` and what you see as a package consumer.

This takes care of `json` and `.d.ts` files; the first because we allow `import` of json files (and we do it in some places, although I think only tests now), the second because tsc does not copy over `.d.ts` files to its output.